bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusSymoInverterHandler.java

```diff
@@ -70,9 +87,81 @@ protected void handleRefresh(FroniusBridgeConfiguration bridgeConfiguration) thr
     @Override
     public void initialize() {
         config = getConfigAs(FroniusBaseDeviceConfiguration.class);
+        FroniusBridgeConfiguration bridgeConfig = getBridge().getConfiguration().as(FroniusBridgeConfiguration.class);
+        if (bridgeConfig.username != null && bridgeConfig.password != null) {
+            batteryControl = new FroniusBatteryControl(httpClient, URI.create("http://" + bridgeConfig.hostname + "/"),
+                    bridgeConfig.username, bridgeConfig.password);
+        }
         super.initialize();
     }
 
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(FroniusSymoInverterActions.class);
+    }
+
+    private @Nullable FroniusBatteryControl getBatteryControl() {
+        if (batteryControl == null) {
+            logger.warn("Battery control is not available. Check the bridge configuration.");
+        }
+        return batteryControl;
+    }
+
+    public void resetBatteryControl() {
+        FroniusBatteryControl batteryControl = getBatteryControl();
+        if (batteryControl != null) {
+            try {
+                batteryControl.reset();
+            } catch (FroniusCommunicationException e) {
+                logger.error("Failed to reset battery control", e);
```

Comment:
```suggestion
                logger.warn("Failed to reset battery control", e);
```

Same for the others in this class. For reference https://www.openhab.org/docs/developer/guidelines.html#f-logging


```
error logging should only be used

to inform the user that something is tremendously wrong in the setup, the system cannot function normally anymore, and there is a need for immediate action.
in case some code fails irrecoverably and the user should report it as a severe bug.
```